packages/@webex/plugin-presence/test/unit/spec/presence-worker.ts

```diff
@@ -0,0 +1,262 @@
+import {assert} from '@webex/test-helper-chai';
+import sinon from 'sinon';
+import Mercury from '@webex/internal-plugin-mercury';
+import Presence from '@webex/plugin-presence';
+import MockWebex from '@webex/test-helper-mock-webex';
+
+import PresenceWorker from '../../../src/presence-worker';
+
+const round = (time) => Math.floor(time / 1000);
+
+describe('presence-worker', () => {
+  describe('PresenceWorker', () => {
+    let webex;
+    let worker;
+    const id = '1234';
+
+    beforeEach(() => {
+      console.log(process.env.WEBEX_CLIENT_ID);
+      webex = new MockWebex({
+        children: {
+          mercury: Mercury,
+          presence: Presence,
+        },
+      });
+      worker = new PresenceWorker();
+      worker.webex = webex;
+    });
+
+    describe('#initialize()', () => {
+      it('requires webex', () =>
+        expect(() => worker.initialize()).toThrow(/Must initialize Presence Worker with webex!/));
+      it('requires webex internal', () =>
+        expect(() => worker.initialize({})).toThrow(/Must initialize Presence Worker with webex!/));
+    });
+
+    // This selection of tests fail due to `webex-core`'s batcher config being missing.
+    describe.skip('#enqueue()', () => {
```

Comment:
skip?

Reply:
Removed the skip. A fix for this was provided by @Shreyas281299 [here](https://github.com/webex/webex-js-sdk/pull/3373#discussion_r1497038638)